### PR TITLE
Trinity: Now handling multiple reads

### PR DESCRIPTION
**Pull request by GallVp**
## PR checklist

Closes #XXX 

- [x] Adapted multiple reads handling strategy from star/align module. This change does not affect existing functionality and tests.
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

**Comment by GallVp** [comment]
Hi @CarsonJM 

Here is an other update. Really appreciate your time.

**Comment by CarsonJM** [comment]
Nice work @GallVp! I think that this works, but am unsure if there is another, preferred way of handling multiple reads. I'll post in the Slack to get some eyes from a more experienced nf-core member!

**Comment by GallVp** [comment]
Thank you @CarsonJM 

I adapted this strategy from STAR/ALIGN https://github.com/nf-core/modules/blob/3a5fef109d113b4997c9822198664ca5f2716208/modules/nf-core/star/align/main.nf#L44

I agree its good to get more feedback.

File: modules/nf-core/trinity/main.nf [hunk 1]
```diff
@@ -9,7 +9,7 @@ process TRINITY {
         'biocontainers/trinity:2.15.1--pl5321h146fbdb_3' }"
 
     input:
-    tuple val(meta), path(reads)
+    tuple val(meta), path(reads, stageAs: "input*/*")
```
Comment (mahesh-panchal) [inline modules/nf-core/trinity/main.nf:12]:
```suggestion
    tuple val(meta), path(reads, stageAs: "input*/*", arity: '1..*')
```

File: modules/nf-core/trinity/main.nf [hunk 2]
```diff
@@ -23,14 +23,17 @@ process TRINITY {
     def args = task.ext.args ?: ''
     def prefix = task.ext.prefix ?: "${meta.id}"
 
+    def reads1 = [], reads2 = []
+    meta.single_end ? [reads].flatten().each{reads1 << it} : reads.eachWithIndex{ v, ix -> ( ix & 1 ? reads2 : reads1) << v }
```
Comment (mahesh-panchal) [inline modules/nf-core/trinity/main.nf:27]:
```suggestion
    meta.single_end ? reads1 = reads : reads.eachWithIndex{ v, ix -> ( ix & 1 ? reads2 : reads1) << v }
```

**Review by mahesh-panchal — COMMENTED** [review]
For better practice, use `arity: '1..*'` to make sure the input is always a `List`. This prevents the need to make a list from reads if it's not one.

An alternative to making reads1 and reads2 is to:
```groovy
--left ${reads.getAt((0..<f.size()).by(2).toList()).join(',')} --right ${reads.getAt((1..f.size()).by(2).toList()).join(',')}
```

**Comment by GallVp** [comment]
Thank you @mahesh-panchal 

I committed your suggestions and all the tests are passing.

**Review by mahesh-panchal — APPROVED** [review]

**Pull request merged**